projects/igniteui-angular/src/lib/grids/tree-grid/tree-grid-integration.spec.ts

```diff
@@ -277,4 +287,231 @@ describe('IgxTreeGrid - Integration', () => {
         });
     });
+
+    describe('Row editing', () => {
+        beforeEach(() => {
+            fix = TestBed.createComponent(IgxTreeGridRowEditingComponent);
+            fix.detectChanges();
+            treeGrid = fix.componentInstance.treeGrid;
+        });
+
+        it('banner has no indentation when editing a parent node.', fakeAsync(() => {
+            // TODO
+            // Verify the overlay has the same width as the row that is edited
+        }));
+
+        it('shows the banner below the edited parent node', fakeAsync(() => {
+            // Collapsed state
+            const grid = fix.componentInstance.treeGrid as IgxTreeGridComponent;
+            grid.collapseAll();
+            fix.detectChanges();
+            verifyBannerPositioning(0);
+
+            // Expanded state
+            grid.expandAll();
+            fix.detectChanges();
+            verifyBannerPositioning(3);
+
+            function verifyBannerPositioning(columnIndex: number) {
+                const cell = grid.getCellByColumn(columnIndex, 'Name');
+                cell.inEditMode = true;
+                tick();
+                fix.detectChanges();
+
+                const editRow = cell.row.nativeElement;
+                const banner = fix.debugElement.query(By.css('.' + CSS_CLASS_BANNER)).nativeElement;
+
+                const bannerTop = banner.getBoundingClientRect().top;
+                const editRowBottom = editRow.getBoundingClientRect().bottom;
+
+                // The banner appears below the row
+                expect(bannerTop).toBeGreaterThanOrEqual(editRowBottom);
+                // No much space between the row and the banner
+                expect(bannerTop - editRowBottom).toBeLessThan(2);
+            }
+        }));
+
+        it('shows the banner below the edited child node', fakeAsync(() => {
+            const grid = fix.componentInstance.treeGrid as IgxTreeGridComponent;
+            grid.expandAll();
+            fix.detectChanges();
+            const cell = grid.getCellByColumn(1, 'Name');
+            cell.inEditMode = true;
+            tick();
+            fix.detectChanges();
+
+            const editRow = cell.row.nativeElement;
+            const banner = fix.debugElement.query(By.css('.' + CSS_CLASS_BANNER)).nativeElement;
+
+            const bannerTop = banner.getBoundingClientRect().top;
+            const editRowBottom = editRow.getBoundingClientRect().bottom;
+
+            // The banner appears below the row
+            expect(bannerTop).toBeGreaterThanOrEqual(editRowBottom);
+            // No much space between the row and the banner
+            expect(bannerTop - editRowBottom).toBeLessThan(2);
+        }));
+
+        xit('shows the banner above the edited parent node if it is the last one', fakeAsync(() => {
+            const grid = fix.componentInstance.treeGrid as IgxTreeGridComponent;
+            grid.height = '200px';
+            fix.detectChanges();
```

Comment:
From my experience, setting grid.height seems to require await or at least a tick().